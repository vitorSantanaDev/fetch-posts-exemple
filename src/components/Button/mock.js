export const buttonProps = {
  text: 'Load More Posts',
  onClick: jest.fn(),
  disabled: false,
};
