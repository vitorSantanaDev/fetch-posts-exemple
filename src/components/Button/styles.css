.buttonContainer {
  height: 100px;
  display: flex;
  align-items: flex-end;
}

.button {
  width: 100%;
  border: none;
  cursor: pointer;
  padding: 16px 32px;
  background-color: #fb1;
  transition: ease-in-out 0.3s;
}

.button:hover {
  opacity: 0.8;
}

.button:disabled {
  cursor: not-allowed;
  background-color: #fea;
}