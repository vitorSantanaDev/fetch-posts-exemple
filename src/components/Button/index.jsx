import React from 'react';
import './styles.css';
import Types from 'prop-types';

const Button = ({ text, onClick, disabled }) => {
  return (
    <div className="buttonContainer">
      <button disabled={disabled} className="button" onClick={onClick}>
        {text}
      </button>
    </div>
  );
};

Button.propTypes = {
  text: Types.string.isRequired,
  onClick: Types.func.isRequired,
  disabled: Types.bool.isRequired,
};
export default Button;
