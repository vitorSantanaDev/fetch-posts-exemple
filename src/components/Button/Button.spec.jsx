import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '.';

const mock = {
  text: 'Load More Posts',
  onClick: jest.fn(),
  disabled: false,
};

describe('<Button/>', () => {
  it('should render the Button with the text "Load More Posts"', () => {
    render(<Button {...mock} />);
    expect.assertions(1);
    const button = screen.getByRole('button', { name: /load more posts/i });
    expect(button).toBeInTheDocument();
  });

  it('should call function on click Button', () => {
    render(<Button {...mock} />);
    const button = screen.getByRole('button', { name: /load more posts/i });
    userEvent.click(button);
    expect(mock.onClick).toHaveBeenCalledTimes(1);
  });

  it('should be disabled when disabled is true', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick} disabled={!mock.disabled} text={mock.text} />);
    const button = screen.getByRole('button', { name: /load more posts/i });
    expect(button).toBeDisabled();
  });

  it('should be enabled when disabled is false', () => {
    render(<Button onClick={mock.onClick} disabled={mock.disabled} text={mock.text} />);
    const button = screen.getByRole('button', { name: /load more posts/i });
    expect(button).toBeEnabled();
  });
});
