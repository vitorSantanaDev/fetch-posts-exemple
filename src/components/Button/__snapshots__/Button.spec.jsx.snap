// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should match snapshot 1`] = `
<div
  class="buttonContainer"
>
  <button
    class="button"
  >
    Load More Posts
  </button>
</div>
`;
